src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,133 @@
+        /// Initialize a new instance of <see cref="SkipQueryOption"/> based on the raw $skiptoken value and
+        /// an EdmModel from <see cref="ODataQueryContext"/>.
+        /// </summary>
+        /// <param name="rawValue">The raw value for $skiptoken query.</param>
+        /// <param name="context">The <see cref="ODataQueryContext"/> which contains the <see cref="IEdmModel"/> and some type information</param>
+        /// <param name="queryOptionParser">The <see cref="ODataQueryOptionParser"/> which is used to parse the query option.</param>
+        public SkipTokenQueryOption(string rawValue, ODataQueryContext context, ODataQueryOptionParser queryOptionParser)
+        {
+            if (context == null)
+            {
+                throw Error.ArgumentNull("context");
+            }
+
+            if (String.IsNullOrEmpty(rawValue))
+            {
+                throw Error.ArgumentNullOrEmpty("rawValue");
+            }
+
+            if (queryOptionParser == null)
+            {
+                throw Error.ArgumentNull("queryOptionParser");
+            }
+
+            RawValue = rawValue;
+            Validator = context.GetSkipTokenQueryValidator();
+            SkipTokenHandler = context.GetSkipTokenHandler();
+            Context = context;
+        }
+
+        /// <summary>
+        /// Gets the raw $skiptoken value.
+        /// </summary>
+        public string RawValue { get; private set; }
+
+        /// <summary>
+        /// Gets and sets the given <see cref="ODataQueryContext"/>.
+        /// </summary>
+        public ODataQueryContext Context { get; set; }
+
+        /// <summary>
+        /// Generates the nextlink value and consumes the skiptoken value.
+        /// </summary>
+        private SkipTokenHandler SkipTokenHandler { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SkipToken Query Validator.
+        /// </summary>
+        public SkipTokenQueryValidator Validator { get; }
+
+        /// <summary>
+        /// Gets or sets the query setting 
+        /// </summary>
+        public ODataQuerySettings QuerySettings { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the list of OrderByNodes for the original query
+        /// </summary>
+        public IList<OrderByNode> OrderByNodes { get; private set; }
+
+        /// <summary>
+        /// Apply the $skiptoken query to the given IQueryable.
+        /// </summary>
+        /// <param name="query">The original <see cref="IQueryable"/>.</param>
+        /// <param name="querySettings">The query settings to use while applying this query option.</param>
+        /// <param name="orderByNodes">Information about the orderby query option.</param>
+        /// <returns>The new <see cref="IQueryable"/> after the skiptoken query has been applied to.</returns>
+        public virtual IQueryable<T> ApplyTo<T>(IQueryable<T> query, ODataQuerySettings querySettings, IList<OrderByNode> orderByNodes)
```

Comment:
>IList<OrderByNode> orderByNodes [](start = 103, length = 31)

why doesn't this take ODataQueryOptions, like the non-generic version?

Reply:
what was the resolution here?

---
In reply to: [292961214](https://github.com/OData/WebApi/pull/1724#discussion_r292961214) [](ancestors = 292961214)